src/server/src/routes/student/index.ts

```diff
@@ -142,6 +143,68 @@ router.get(
     }
 );
 
+/******************************************************************************
+ *            GET Request - Search - /api/student/search"
+ ******************************************************************************/
+
+interface StudentSearchRequest extends Request {
+    query: {
+        firstName?: string;
+        lastName?: string;
+        email?: string;
+        sid?: string;
+        departmentIds?: string[];
+        classStandings?: classStandings[];
+        page: string;
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/search',
+    passport.authenticate('jwt', { session: false }),
+    validationMiddleware({ querySchema: studentSearchSchema }),
+    async (req: StudentSearchRequest, res: Response) => {
+        const { departmentIds, page, numOfItems } = req.query;
+        let { firstName, lastName, email, sid, classStandings } = req.query;
+
+        if (!firstName) firstName = "";
+        if (!lastName) lastName = "";
+        if (!email) email = "";
+        if (!sid) sid = "";
+        var departmentIdInts = [];
```

Comment:
var -> const. Note avoid using var, they are bad practice in typescript

Reply:
But I need to push things into departmentIdInts. Can I push things into const array?

Reply:
Done.